paymentintent.go

```diff
@@ -0,0 +1,175 @@
+}
+
+// UnmarshalJSON handles deserialization of a PaymentIntentSourceAction.
+// This custom unmarshaling is needed because the specific
+// type of for `value` it refers to is specified in the `type` property
+func (s *PaymentIntentSourceAction) UnmarshalJSON(data []byte) error {
+	type paymentIntentSourceAction PaymentIntentSourceAction
+	var v paymentIntentSourceAction
+	if err := json.Unmarshal(data, &v); err != nil {
+		return err
+	}
+
+	var err error
+	*s = PaymentIntentSourceAction(v)
+	s.Value = &PaymentIntentSourceActionValue{}
+
+	// Unmarshal data a second time so that we can get the raw bytes for the
+	// `value` field
+	var rawObject map[string]*json.RawMessage
+	if err := json.Unmarshal(data, &rawObject); err != nil {
+		return err
+	}
+
+	switch s.Type {
+	case PaymentIntentNextActionAuthorizeWithURL:
+		err = json.Unmarshal(*rawObject["value"], &s.Value.AuthorizeWithURL)
+	}
+
+	return err
+}
+
+type PaymentIntentTransferData struct {
+	Amount int64 `json:"amount"`
+}
+
+// Payout is the resource representing a Stripe payout.
+// For more details see https://stripe.com/docs/api#payouts.
+type PaymentIntent struct {
+	AllowedSourceTypes  []string                        `json:"allowed_source_types"`
+	Amount              int64                           `json:"amount"`
+	AmountCapturable    int64                           `json:"amount_capturable"`
+	AmountReceived      int64                           `json:"amount_received"`
+	Application         *Application                    `json:"application"`
+	ApplicationFee      int64                           `json:"application_fee"`
+	CanceledAt          int64                           `json:"canceled_at"`
+	CaptureMethod       PaymentIntentCaptureMethod      `json:"capture_method"`
+	Charges             *ChargeList                     `json:"charges"`
+	ClientSecret        string                          `json:"client_secret"`
+	ConfirmationMethod  PaymentIntentConfirmationMethod `json:"confirmation_method"`
+	Created             int64                           `json:"created"`
+	Currency            string                          `json:"currency"`
+	Customer            *Customer                       `json:"customer"`
+	Description         string                          `json:"description"`
+	Livemode            bool                            `json:"livemode"`
+	ID                  string                          `json:"id"`
+	Metadata            map[string]string               `json:"metadata"`
+	NextSourceAction    *PaymentIntentSourceAction      `json:"next_source_action"`
+	OnBehalfOf          *Account                        `json:"on_behalf_of"`
+	ReceiptEmail        string                          `json:"receipt_email"`
+	ReturnURL           string                          `json:"return_url"`
+	Shipping            ShippingDetails                 `json:"shipping"`
+	Source              *Source                         `json:"source"`
```

Comment:
Since PaymentIntent will only support Sources and nothing else, I think it's the right call to use `Source` here.